src/s3path.jl

```diff
@@ -40,6 +40,36 @@ function S3Path()
     )
 end
 
+function S3Path(
+    bucket::AbstractString,
+    key::AbstractString;
+    isdirectory::Bool=false,
+    config::AWSConfig=aws_config(),
+)
+    return S3Path(
+        Tuple(filter!(!isempty, split(key, "/"))),
+        "/",
+        strip(startswith(bucket, "s3://") ? bucket : "s3://$bucket", '/'),
+        isdirectory,
+        config,
+    )
+end
+
+function S3Path(
+    bucket::AbstractString,
+    key::AbstractPath;
+    isdirectory::Bool=false,
+    config::AWSConfig=aws_config(),
+)
+    return S3Path(
+        key.segments,
+        "/",
+        strip(startswith(bucket, "s3://") ? bucket : "s3://$bucket", '/'),
```

Comment:
maybe extract this into a function, as a way to give it a clear name.
`normalize_bucket_name(bucket)`

Reply:
Sure